### ISSUE TITLE
Doxywizard 1.8.19 (Windows): Source code directory seems to be ignored

### ISSUE THREAD

**Issue by Roxxorfreak** [doxygen/doxygen#7951]
# Problem

Using Doxywizard 1.8.19 under Windows 10, running doxygen, no source files are found (any more),
The html file structure is created, but no entities are documented.
Doxyfiles from previous versions do not work anymore.

# Settings

On the "Project" page in the Wizard, both "Working directory" and "Source code directory" are specified (see image).
Working directory: "D:/Projects/hedit/tools/doxygen"
Source code directory: "../../src" (should expand to "D:/Projects/hedit/src")

![error1](https://user-images.githubusercontent.com/66142617/89772589-8ec64f00-db02-11ea-9079-5cb2d0dcbee8.PNG)

# Result
The generated html documentation is empty, since no source files are found.
As marked in the image, Doxygen seems to use the "working directory" to look for source files.

![error2](https://user-images.githubusercontent.com/66142617/89772668-b6b5b280-db02-11ea-9c62-3c0571c89e2f.PNG)

# Remarks 
The same Doxyfile is working using Doxygen 1.8.18.

![error3](https://user-images.githubusercontent.com/66142617/89773946-2e84dc80-db05-11ea-8c2a-ddd7cb1da8c9.PNG)

**Comment by albert-github** [comment]
Description is clear but an example directory structure would have been nice, example: [example.tar.gz](https://github.com/doxygen/doxygen/files/5050580/example.tar.gz)

First idea is that in one or another way the current directory is lost, even an absolute path does not work.
Looking at the code there is only one change in the doxywizard between 1.8.18 and 1.8.19 but this is not relevant for the mentioned problem (it is about displaying the version, `doxywizard --version`). This has to be investigated...

As a workaround it is always possible to start doxygen from the mentioned directory.

**Comment by Roxxorfreak** [comment]
Sorry, you can just use this repo: https://github.com/Roxxorfreak/HEdit.git
The Doxyfile is located in /tools/doxygen/

I did some additional tests and the problem only occurs when running doxygen through the Doxywizard. 
Starting it from command line (with the same config file), everything works as expected...

Attached is the log file when starting Doxygen from command line ("doxygen.exe Doxyfile"). As you can see, this works...
[output_from_1_8_19.txt](https://github.com/doxygen/doxygen/files/5050928/output_from_1_8_19.txt)

**Comment by albert-github** [comment]
With the real small example as I attached in comment: https://github.com/doxygen/doxygen/issues/7951#issuecomment-671316693 I was already able to reproduce the problem.

The https://github.com/Roxxorfreak/HEdit.git is a bit less suitable as:
- it is not really minimal
- it is not persistent (not that relevant in this case though).

It looks like the settings don't arrive with the doxygen executable (they are send to stdin of doxygen) as doxygen is started as `doxygen -` , though `type Doxyfile | doxygen -` does work, still investigating....

**Comment by Roxxorfreak** [comment]
If I can test anything, just ask :-)

**Comment by albert-github** [comment]
My first conclusion that `type Doxyfile | doxygen -` works is not correct (the fact that it did work for me was because of the small test Doxyfile I used  :-( ).

After about 4096 characters the rest of the input from the Doxyfile is discarded in the 1.8.19 doxygen version, in the doxygen 1.8.18 version the entire file is read, so it looks like the problem is coming from a buffer size, on the doxygen side (not doxywizard!), in case the Doxyfile is read from standard input.

**Comment by Roxxorfreak** [comment]
Thanks, nice debugging work! So, currently the only solution is: Using the command line :-)

**Comment by albert-github** [comment]
I did some further searching and the problem is caused in the function `static QCString configFileToString(const char *name)` on `configimpl.l`. This function did not change between the 1.8.18 and 1.8.19 version but when increasing the `bsize` from 4096 to e.g. 400096 everything went smooth.

Checking the different commits the commit:
```
Commit: 84e15087843c96ba73d1ec63d97e6d28a12d9766 [84e1508]
Parents: 27fd76fde3, 1fa64cc88d
Date: Wednesday, June 10, 2020 8:36:58 PM
Merge branch 'refactor_qcstring'
```
is the commit causing the problems.

**Comment by doxygen** [comment]
@albert-github Can you check if the problem is fixed with the commit I just pushed? The buffer size was 1 byte too small it seems. The `QCString` implementation is since version 1.8.19 based on `std::string`, which may explain why this is now revealed while it was not flagged as an issue before.

**Comment by albert-github** [comment]
It looks to me that the problem has been solved., I now see  `.../hedit/src`.

**Comment by MarkShackman** [comment]
Just to note that this bug has broken all our builds (as we pipe both the original and a variant doxyfile into doxygen) so a swift release of the fixed version would be appreciated.

Mark

**Comment by ReggThims** [comment]
Just spent the last 3 hours trying to understand what is going on here... so pleased to find this thread.
I can make my project read the source files ONLY if I set the working directory to be the folder holding the source, which breaks all my configuration files.
Also, the IMAGE_PATH folder seems to be ignored... all my images are marked as not found.
I suspect that all pathing is broken in 1.8.19.

**Comment by albert-github** [comment]
@ReggThims 

- Did you build through the doxywizard? If so you can either use the way you describe () or use doxygen directly (`doxygen Doxyfile`) on your "normal directory.
- "I suspect that all pathing is broken in 1.8.19." With paths you see it easiest but in fact all settings, when using the doxygen wizard, from `OUTPUT_LANGUAGE` are broken. When just piping you might be a bit luckier (and get a bit further, but also here there will be parts broken).

**Comment by ReggThims** [comment]
@albert-github 
Thanks for the reply... I can survive with the command line... any idea of when the fix will be released?

**Comment by doxygen** [comment]
@ReggThims I'll see if I can make a bug fix release soon for this.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.20.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
